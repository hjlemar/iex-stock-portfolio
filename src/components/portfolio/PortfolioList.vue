<template>
  <v-layout row wrap>
    <v-flex xs12 v-for="(portfolio,i) in portfolios" :key="i">
      <app-portfolio-list-item
        :portfolio="portfolio"
        @click.native="openPortfolio(portfolio)"
      ></app-portfolio-list-item>
      <v-divider v-if="i < portfolios.length"></v-divider>
    </v-flex>
  </v-layout>
</template>

<script>
import PortfolioListItem from './PortfolioListItem';

export default {
  props: ['portfolios'],
  components: {
    appPortfolioListItem: PortfolioListItem,
  },
  methods: {
    openPortfolio(portfolio) {
      this.$router.push({ path: `/portfolio/${portfolio}` });
    },
  },
};
</script>

<style>

</style>
