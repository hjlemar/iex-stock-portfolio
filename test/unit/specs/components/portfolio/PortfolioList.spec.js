import { shallow, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import PortfolioList from '@/components/portfolio/PortfolioList';
import Vuetify from 'vuetify';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('PortfolioList.vue', () => {
  let store;
  beforeEach(() => {
    store = {};
  });

  it('it matches its snapshot with no portfolios', () => {
    const wrapper = shallow(PortfolioList, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('it matches its snapshot with two Portfolios', () => {
    const wrapper = shallow(PortfolioList, {
      store,
      localVue,
      propsData: {
        portfolios: ['A', 'B'],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
