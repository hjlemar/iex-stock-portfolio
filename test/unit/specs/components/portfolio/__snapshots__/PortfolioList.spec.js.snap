// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioList.vue it matches its snapshot with no portfolios 1`] = `undefined`;

exports[`PortfolioList.vue it matches its snapshot with two Portfolios 1`] = `
<div class="layout row wrap">
    <div class="flex xs12">
        <!---->
        <hr class="divider">
    </div>
    <div class="flex xs12">
        <!---->
        <hr class="divider">
    </div>
</div>
`;
